package.json

```diff
@@ -47,7 +47,8 @@
     "chromatic:s2": "NODE_ENV=production CHROMATIC=1 chromatic --buildScriptName='build:chromatic-s2' --project-token=$RAINBOW_CHROMATIC_PROJECT_TOKEN",
     "merge:css": "babel-node --presets @babel/env ./scripts/merge-spectrum-css.js",
     "release": "lerna publish from-package --yes",
-    "publish:nightly": "lerna publish -y --canary --preid nightly --dist-tag=nightly --exact --force-publish='*' --no-push --no-verify-access",
+    "version:nightly": "yarn workspaces foreach --all --no-private -t version -d 3.0.0-nightly-$(git rev-parse --short HEAD)-$(date +'%y%m%d') && yarn version apply --all",
```

Comment:
I think this has a couple of advantages compared to our current version scheme for nightly
```
@adobe/react-spectrum: 3.0.0-nightly.3102
react-aria-components: 1.3.4-nightly.5030
```
would become
```
@adobe/react-spectrum: 3.0.0-nightly-7ae45ad-20240829
react-aria-components: 3.0.0-nightly-7ae45ad-20240829
```
This would take out a lot of guess work when trying to align nightly versions that aren't on the nightly tag.
In addition, we now know exactly which commit the nightly is built off of, and we have the date in an easy to read place. The UI on npmjs isn't great for finding this information, so I think it's helpful to include it. 

The downside to this is this might be confusing to users who primarily use RAC which is still in major version 1.

I took the idea for this scheme from React https://www.npmjs.com/package/react?activeTab=versions

Reply:
Don't we have a package we bumped to v4 as well as RAC on v1? Would they get squished, or did we rename with the version bump?

I like the commit and date part. I feel like we're throwing out SemVer and always showing zeros for minot and patch could be confusing too. Would it be less confusing if we only show the correct major (v1, v3, or v4) or stick x's in the minor and patch?

Reply:
> Don't we have a package we bumped to v4 as well as RAC on v1? Would they get squished, or did we rename with the version bump?

Yes, that's why I showed two cases where the major version was different. They all become 3.

> I like the commit and date part. I feel like we're throwing out SemVer and always showing zeros for minot and patch could be confusing too. Would it be less confusing if we only show the correct major (v1, v3, or v4) or stick x's in the minor and patch?

Nightly doesn't truly follow semver anyways since if you use it for one dependency, you must use it for all dependencies or get duplicates. As such, I think the best thing is to make it easy to match nightly versions across all dependencies. Including the date and git commit allow you to quickly narrow down so you know if you're including a change you are interested in as well. Compared to a random increasing integer, even with the semver fields intact, I think my proposal makes it faster to find what you want.

You cannot put an 'x' in any of the semver major/minor/patch fields, that's invalid. See https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions